cdap-ui/app/features/workflows/controllers/tabs/runs/tabs/status-ctrl.js

```diff
@@ -5,18 +5,39 @@ angular.module(PKG.name + '.feature.workflows')
           appId: $state.params.appId,
           workflowId: $state.params.programId,
           scope: $scope
-        };
+        },
+        match,
+        runparams;
 
     if ($state.params.runid) {
-      var match = filterFilter($scope.RunsController.runs, {runid: $state.params.runid});
+      match = filterFilter($scope.RunsController.runs, {runid: $state.params.runid});
       if (match.length) {
         $scope.RunsController.runs.selected = match[0];
       }
     }
 
     var vm = this;
+    vm.runStatus = null;
 
     vm.data = {};
+
+    runparams = angular.extend(
+      {
+        runId: $scope.RunsController.runs.selected.runid
+      }, params);
+
+    if ($scope.RunsController.runs.length) {
+      myWorkFlowApi
+        .pollRunDetail(runparams)
```

Comment:
@elfenheart Yes This is what 2nd TODO comment is talking about. Pushed out this PR so that it doesn't become too big to review. Will add that commit later tonight or tomorrow morning.